packages/mui-base/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -338,33 +338,105 @@ export default function useAutocomplete<
   FreeSolo extends boolean | undefined = false,
 >(
   props: UseAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
-): {
+): UseAutocompleteReturnValue<T, Multiple, DisableClearable, FreeSolo>;
+
+export interface UseAutocompleteRenderedOption<T> {
+  option: T;
+  index: number;
+}
+
+export interface UseAutocompleteReturnValue<
+  T,
+  Multiple extends boolean | undefined = false,
+  DisableClearable extends boolean | undefined = false,
+  FreeSolo extends boolean | undefined = false,
+> {
+  /**
+   * Resolver for the root slot's props.
+   * @param externalProps props for the root slot
+   * @returns props that should be spread on the root slot
+   */
   getRootProps: (externalProps?: any) => React.HTMLAttributes<HTMLDivElement>;
+  /**
+   * Resolver for the input element's props.
+   * @returns props that should be spread on the input element
+   */
   getInputProps: () => React.InputHTMLAttributes<HTMLInputElement>;
-  // We pass `getInputLabelProps()` to `@mui/material/InputLabel` which does not implement HTMLLabelElement#color.
+  /**
+   * Resolver for the input label element's props.
+   * @returns props that should be spread on the input label element
+   */
   getInputLabelProps: () => Omit<React.HTMLAttributes<HTMLLabelElement>, 'color'>;
+  /**
+   * Resolver for the `clear` button element's props.
+   * @returns props that should be spread on the *clear* button element
+   */
   getClearProps: () => React.HTMLAttributes<HTMLButtonElement>;
+  /**
+   * Resolver for the popup icon's props.
+   * @returns props that should be spread on the popup icon
+   */
   getPopupIndicatorProps: () => React.HTMLAttributes<HTMLButtonElement>;
+  /**
+   * A tag props getter.
+   */
   getTagProps: AutocompleteGetTagProps;
+  /**
+   * Resolver for the listbox component's props.
+   * @returns props that should be spread on the listbox component
+   */
   getListboxProps: () => React.HTMLAttributes<HTMLUListElement>;
-  getOptionProps: ({
-    option,
-    index,
-  }: {
-    option: T;
-    index: number;
-  }) => React.HTMLAttributes<HTMLLIElement>;
+  /**
+   * Resolver for the rendered option element's props.
+   * @param renderedOption option rendered on the Autocomplete
+   * @returns props that should be spread on the li element
+   */
+  getOptionProps: (
+    renderedOption: UseAutocompleteRenderedOption<T>,
+  ) => React.HTMLAttributes<HTMLLIElement>;
+  /**
+   * Id for the Autocomplete.
+   */
   id: string;
+  /**
+   * The input value.
+   */
   inputValue: string;
+  /**
+   * The value of the autocomplete.
+   */
   value: AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>;
+  /**
+   * If `true`, the component has valid values selected.
```

Comment:
I think this should be:
```suggestion
   * If `true`, the component input has some values.
```
Because in `freeSolo` mode, it can be any value.

Reply:
Good call-out, have amended my PR